packages/filter-workspace-packages/src/index.ts

```diff
@@ -66,13 +70,54 @@ export async function filterPkgsBySelectorObjects<T> (
     selectedProjectsGraph: PackageGraph<T>
     unmatchedFilters: string[]
   }> {
-  const { graph } = createPkgGraph<T>(pkgs, { linkWorkspacePackages: opts.linkWorkspacePackages })
-  if (packageSelectors?.length) {
-    return filterGraph(graph, packageSelectors, {
-      workspaceDir: opts.workspaceDir,
-      testPattern: opts.testPattern,
+  packageSelectors = packageSelectors.filter((selector) => !selector.followProdDepsOnly)
+  const prodPackageSelectors = packageSelectors.filter((selector) => selector.followProdDepsOnly)
+
+  if ((packageSelectors.length > 0) || (prodPackageSelectors.length > 0)) {
+    let filteredGraph = Promise.resolve<{
+      selectedProjectsGraph: PackageGraph<T>
+      unmatchedFilters: string[]
+    }>({
+      selectedProjectsGraph: {},
+      unmatchedFilters: [],
+    })
```

Comment:
there is no need in default value, just use this when reading it:

```
...(await prodFilteredGraph)?.selectedProjectsGraph,
```

Reply:
I'm getting `Variable 'prodFilteredGraph' is used before being assigned.` when I remove the default value.

Reply:
Trie to assign a null or undefined then